lucene/core/src/test/org/apache/lucene/store/TestLockFactoriesMultiJVM.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.store;
+
+import java.lang.ProcessBuilder.Redirect;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.lucene.util.LuceneTestCase;
+import org.apache.lucene.util.NamedThreadFactory;
+import org.apache.lucene.util.SuppressForbidden;
+
+public class TestLockFactoriesMultiJVM extends LuceneTestCase {
```

Comment:
I would maybe mark this as belonging to Slow group? Just a thought.

Reply:
FYI I ran it locally on a 2-core laptop. The tests are a bit slow but not the slowest in lucene-core.

The slowest tests (exceeding 500 ms) during this run:
   2.76s TestIndexWriter.testMaxCompletedSequenceNumber (:lucene:core)
   1.96s TestNRTCachingDirectory.testNRTAndCommit (:lucene:core)
   1.69s TestIndexWriter.testThreadInterruptDeadlock (:lucene:core)
   1.60s TestShardSearching.testSimple (:lucene:core)
   1.55s TestStandardAnalyzer.testLargePartiallyMatchingToken (:lucene:core)
   1.46s TestTieredMergePolicy.testSimulateAppendOnly (:lucene:core)
   **1.32s TestLockFactoriesMultiJVM.testNativeFSLockFactory (:lucene:core)**
   **1.23s TestLockFactoriesMultiJVM.testSimpleFSLockFactory (:lucene:core)**
   1.17s TestLevenshteinAutomata.testLev2 (:lucene:core)
   1.15s TestIndexSorting.testConcurrentUpdates (:lucene:core)

Reply:
It will be sligtly faster soon, but can be tuned. But marking it as `@Slow` is a bad idea because this is a very important test! Especially to collect information from users, if it really works everywhere.

To really do extensive testing do `-Dtests.nightly=true`, then it runs about 2 minutes.